yax/state/artifact_map.py

```diff
@@ -70,31 +147,94 @@ def _declare_new_artifact(self, run_id, bound_artifact):
         self._insert_into("Artifact_Run",
                           {'run_id': run_id, 'artifact_id': artifact_id})
 
-    def create_run(self, config):
+    def declare_run(self, config):
+        """Create or find a run for the given config.
+
+        Parameters
+        ----------
+        config : dict of dicts
+            Maps section name to dict of parameters and their arguments.
+
+        Returns
+        -------
+        (int, str)
```

Comment:
The return is technically a tuple, so there aren't really names for the caller's perspective.